include/Jit/utility.h

```diff
@@ -20,89 +20,159 @@
 #include <cassert>
 #include <memory>
 
+#include "cor.h"
+
 #include "llvm/Support/Atomic.h"
+#include "llvm/ADT/STLExtras.h"
 #include "llvm/Config/llvm-config.h"
 
-/// \brief MethodName struct representing a particular method on a type.
+///////using namespace llvm;
+
+// We overload the MethodID.NumArgs field to denote the status of any
+// particular parse:
+//   BAD    => the string did not conform to any legal pattern.
```

Comment:
I think we should not overload BAD/EMPTY state on NumArgs field because they are not related.
It is better to avoid the bad/empty states. 
For example, Methodset::insert can parse the incoming string first, and only construct Method ID objects for valid names.